src/test/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtilTest.java

```diff
@@ -95,9 +95,11 @@ public void testGetTableSchema() {
     Schema.Field cdapSubStringField = Schema.Field.of("subStringField",
         Schema.nullableOf(Schema.of(Schema.Type.STRING)));
     Schema.Field cdapSubRecordField = Schema.Field.of("subStructure",
-        Schema.nullableOf(Schema.recordOf("structure.subStructure", cdapSubStringField)));
+        Schema.nullableOf(Schema.recordOf("structure.subStructure" +
+                                            Schema.recordOf(cdapSubStringField).getRecordName(), cdapSubStringField)));
     Schema.Field cdapRecordField = Schema.Field.of("structure",
-        Schema.nullableOf(Schema.recordOf("structure", cdapSubRecordField)));
+        Schema.nullableOf(Schema.recordOf("structure" +
+          Schema.recordOf(cdapSubRecordField).getRecordName(), cdapSubRecordField)));
     cdapFieldList.add(cdapRecordField);
 
```

Comment:
For such a large change, I would like additional tests added to validate that this doesn't change existing behaviors.

Reply:
Added the test to check depth validation as discussed offline.